code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -134,6 +152,16 @@
 		var/datum/action/A=X
 		A.UpdateButtonIcon()
 
+/obj/item/clothing/head/helmet/space/plasmaman/AltClick(mob/user)
+	..()
+	. = COMSIG_CLICK_ALT
+	if(!attached_hat)
+		return
+	user.put_in_active_hand(attached_hat)
+	to_chat(user, "<span class='notice'>You removed [attached_hat.name] from helmet!</span>")
+	attached_hat = null
```

Comment:
ok, so, idk dm much, but it seems that if your active hand is full and you take it off, it doesn't put it in your active hand but still sets attached_hat to null